mx.py

```diff
@@ -2096,15 +2096,15 @@ def _get_path_in_cache(name, sha1, urls, ext=None):
 
 def download_file_exists(urls):
     '''
-    Returns true if the given urls really exist on the server.
+    Returns true if one of the given urls denotes an existing resource.
     '''
-    _, binDir = _compile_mx_class('URLConnectionDownload')
-    command = [get_jdk(tag=DEFAULT_JDK_TAG).java, '-cp', _cygpathU2W(binDir), 'URLConnectionDownload']
-    command += ['--no-progress', '--skip-download']
-    command += urls
-    def devnull(s):
-        pass
-    return run(command, err=devnull, nonZeroIsFatal=False) == 0
+    for url in urls:
+        try:
+            urllib2.urlopen(url, timeout=0.5).close()
```

Comment:
I use download_file_exists to probe if the maven artifacts are already existing. So if the probing takes longer than the writing (in the common case) then it kinda defeats the purpose.